operator/pkg/controller/cassandradatacenter/cassandradatacenter_controller.go

```diff
@@ -146,12 +146,36 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 			return requests
 		})
 
+	nodeTaintsChangedPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			nodeOld, ok1 := e.ObjectOld.(*corev1.Node)
+			nodeNew, ok2 := e.ObjectNew.(*corev1.Node)
+
+			if !ok1 || !ok2 {
+				log.Error(nil, "Failed to cast update.Event objects to type Node", "objectOld", e.ObjectOld, "objectNew", e.ObjectNew)
+				return true
+			}
+
+			if nodeOld == nil && nodeNew == nil {
+				return false
```

Comment:
I think if ok1 and ok2 are both true, these nodes must not be nil?

Reply:
the interface value might be (nil, corev1.Node). In which case it can be cast to corev1.Node and still be nil with ok1 and ok2 being true.